packages/app/src/cli/models/extensions/specifications/flow_template.ts

```diff
@@ -0,0 +1,82 @@
+// import {createUIExtensionSpecification} from '../ui.js'
+import {BaseSchema} from '../schemas.js'
+import {createExtensionSpecification} from '../specification.js'
+import {loadLocalesConfig} from '../../../utilities/extensions/locales-configuration.js'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {zod} from '@shopify/cli-kit/node/schema'
+import {AbortError} from '@shopify/cli-kit/node/error'
+import {glob} from '@shopify/cli-kit/node/fs'
+import fs from 'fs'
+
+const FlowTemplateExtensionSchema = BaseSchema.extend({
+  name: zod.string(),
+  description: zod.string(),
+  extensions: zod
+    .array(
+      zod.object({
+        handle: zod.string(),
+        type: zod.literal('flow_template'),
+        title: zod.string(),
+        description: zod.string(),
+        categories: zod.array(zod.string()),
+        require_app: zod.boolean(),
+        visible: zod.boolean(),
+        enabled: zod.boolean(),
+      }),
+    )
+    .min(1)
+    .max(1),
+})
+
+const spec = createExtensionSpecification({
+  identifier: 'flow_template',
+  schema: FlowTemplateExtensionSchema,
+  singleEntryPath: false,
+  appModuleFeatures: (_) => ['bundling'],
+  deployConfig: async (config, directory) => {
+    // supporting one extension for now in implementation
+    const extension = config.extensions[0]
+    if (!extension) {
+      throw new AbortError(`Missing extension in ${directory} `, 'Make sure you have defined at least one extension.')
+    }
+    if (config.extensions.length > 1) {
+      throw new AbortError(
+        `More than one extension found in ${directory} `,
+        'Make sure you have only one extension defined. This will be expanded to support more in the future.',
+      )
+    }
```

Comment:
Do we need this now with the `min`/`max` changes with the zod schema?

Reply:
The following check is nice for typing (since otherwise Typescript considers that `extension` can  be `undefined`)

```
if (!extension) {
  throw new AbortError(`Missing extension in ${directory} `, 'Make sure you have defined at least one extension.')
}
```

but the check for `config.extensions.length > 1` isn't necessary anymore IMO.

Reply:
Since its 2 different types of validation, typescript will still yell at us saying extension could be null in this case. It doesn't hurt to have the first extra check even if it is slightly redundant at the moment.

Reply:
Ah you know what I had forgotten that I had removed that check in favour of coercing the typing of `extension` [here](https://github.com/Shopify/cli/pull/2139#discussion_r1254803864).